Desktop_Interface/functiongencontrol.cpp

```diff
@@ -5,126 +5,56 @@
 functionGenControl::functionGenControl(QWidget *parent) : QLabel(parent)
 {
     this->hide();
-    samples_CH1 = (unsigned char *) malloc(0);
```

Comment:
I can only imagine why 2016 me thought it was a good to malloc a 0 byte array?

Reply:
Other functions called free() on samples_CH1 and samples_CH2 first thing, it seems this was in place to have something to free on the first call. But since free(NULL) is a no-op, setting it to NULL is perfectly valid, which is what i changed it to.